src/DTO/BaseEventDTO.php

```diff
@@ -4,73 +4,73 @@
 
 namespace Abrouter\Client\DTO;
 
-class EventDTO
+class BaseEventDTO
 {
     /**
-     * @var string|null
+     * @var string
      */
-    private $temporaryUserId;
+    private string $event;
 
     /**
      * @var string|null
      */
-    private $userId;
+    private ?string $temporaryUserId;
 
     /**
-     * @var string
+     * @var string|null
      */
-    private $event;
+    private ?string $userId;
 
     /**
      * @var string|null
      */
-    private $referrer;
+    private ?string $referrer;
 
     /**
      * @var string|null
      */
-    private $tag;
+    private ?string $tag;
 
     /**
      * @var array|null
      */
-    private $meta;
+    private ?array $meta;
 
     /**
      * @var string|null
      */
-    private $ip;
+    private ?string $ip;
 
     /**
      * @var string|null
      */
-    private $created_at;
+    private ?string $created_at;
 
     /**
-     * EventDTO constructor.
+     * BaseEventDTO constructor.
      *
+     * @param string      $event
      * @param string|null $temporaryUserId
      * @param string|null $userId
-     * @param string      $event
      * @param string|null $tag
      * @param string|null $referrer
      * @param array|null  $meta
      * @param string|null $ip
      * @param string|null $created_at
      */
     public function __construct(
+        string $event,
```

Comment:
No-no-no! We can't change it already without releasing the major version. Somebody can already use it. We don't plan to change it at all. We plan to create the builder for this class.